app/controllers/crops_controller.rb

```diff
@@ -127,34 +89,25 @@ def create
 
     if current_member.role? :crop_wrangler
       @crop.creator = current_member
-      success_msg = "Crop was successfully created."
     else
       @crop.requester = current_member
       @crop.approval_status = "pending"
-      success_msg = "Crop was successfully requested."
     end
 
-    respond_to do |format|
-      if @crop.save
-        params[:alt_name].each do |index, value|
-          create_name('alternate', value)
-        end
-        params[:sci_name].each do |index, value|
-          create_name('scientific', value)
-        end
-        unless current_member.role? :crop_wrangler
-          Role.crop_wranglers.each do |w|
-            Notifier.new_crop_request(w, @crop).deliver_now!
-          end
+    if @crop.save
```

Comment:
I reckon this should be in a transaction -- so you get the whole crop data saved, never half